src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -584,6 +584,12 @@ private void CompileNamedType(NamedTypeSymbol containingType)
                 }
             }
 
+            var properties = members.OfType<SourcePropertySymbolBase>();
+            foreach (var property in properties)
+            {
+                property.MarkBackingFieldAsCalculated();
```

Comment:
>property.MarkBackingFieldAsCalculated(); [](http://example.com/codeflow?start=16&length=40)

I think we should take ```_filterOpt``` into account. This can be called only when all accessors pass the filter. Alternatively we can skip the loop if there is a filter.

Reply:
Before fixing this, create a unit test that exposes the problem. I think one way to have a filter is to request diagnostics for specific SyntaxTree. In this scenario ```MarkBackingFieldAsCalculated``` shouldn't be called for a property declared in a partial declaration in a different SyntaxTree because its accessors were not compiled in the loop above. It should be possible to observe the problem through ```FieldKeywordBackingField``` property, or by an unexpected behavior during binding of the accessors (the ```field``` symbol isn't found, etc.).